File: test_ws.py
import websocket # type: ignore

def test_ws(url, origin):
    print(f"Connecting to {url} with Origin: {origin}")
    try:
        ws = websocket.create_connection(url, origin=origin, timeout=5)
        print("Connected successfully!")
        ws.close()
    except Exception as e:
        print(f"Connection failed: {e}")

try:
    test_ws("wss://api.guig.dev/v1/asr/stream", "https://ultrablabla.guig.dev")
    test_ws("wss://api.guig.dev/v1/voice/stream", "https://ultrablabla.guig.dev")
except Exception as err:
    print(err)

File: test_ws.ts
async function testSocket(url: string, origin: string) {
  console.log(`\n--- Testing ${url} (Origin: ${origin}) ---`);
  return new Promise((resolve) => {
    try {
      const ws = new WebSocket(url, {
        headers: {
          Origin: origin,
        }
      } as any);

      const timer = setTimeout(() => {
        console.error('TIMEOUT after 5s');
        ws.close();
        resolve(false);
      }, 5000);

      ws.onopen = () => {
        console.log('WS OPENED!');
        clearTimeout(timer);
        if (url.includes('voice')) {
          ws.send(JSON.stringify({ type: 'chat', text: 'Bonjour', voice: 'guillaume' }));
        } else if (url.includes('asr')) {
          ws.send(JSON.stringify({ type: 'start', language: 'fr-CA', sample_rate: 16000 }));
        }
      };

      ws.onmessage = (ev) => {
        console.log('WS MESSAGE RECEIVED:', ev.data);
      };

      ws.onerror = (err: any) => {
        console.error('WS ERROR:', err.message || err);
      };

      ws.onclose = (ev) => {
        console.log(`WS CLOSED: code=${ev.code}, reason=${ev.reason}`);
        clearTimeout(timer);
        resolve(true);
      };
    } catch (e) {
      console.error('Exception:', e);
      resolve(false);
    }
  });
}

await testSocket('wss://api.guig.dev/v1/asr/stream', 'https://ultrablabla.guig.dev');
await testSocket('wss://api.guig.dev/v1/voice/stream', 'https://ultrablabla.guig.dev');
